src/Psr11ConnectionRegistry.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Doctrine\Bundle\DBALBundle;
+
+use Doctrine\DBAL\Connection;
+use Psr\Container\ContainerInterface;
+
+class Psr11ConnectionRegistry implements ConnectionRegistry
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @var string
+     */
+    private $defaultConnectionName;
+
+    /**
+     * @var string[]
+     */
+    private $connectionNames;
+
+    public function __construct(ContainerInterface $container, string $defaultConnectionName, array $connectionNames)
+    {
+        $this->container = $container;
+        $this->defaultConnectionName = $defaultConnectionName;
+        $this->connectionNames = $connectionNames;
+    }
+
+    public function getDefaultConnectionName(): string
+    {
+        return $this->defaultConnectionName;
+    }
+
+    public function getConnection(?string $name = null): Connection
```

Comment:
should this return type hint be nullable? what should happen if there is no connection with the given name? Return `null` or throw `\InvalidArgumentException`?

Reply:
Ok it seems `\InvalidArgumentException` would be in line with what happens already in the ManagerRegistry:

https://github.com/doctrine/persistence/blob/1.3.x/lib/Doctrine/Persistence/AbstractManagerRegistry.php#L95